### PR TITLE
chore: update onpointerevent in SDK

### PR DESCRIPTION
**Pull request by pravusjif**
* Updated OnPointerDown, OnPointerUp and OnClick component models in kernel
* Implemented new params in unity side
* Updated kernel-unity data interaction to use the new model and behave accordingly
* Updated cursor images

### CODE REVIEW

File: kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts [hunk 1]
```diff
@@ -92,52 +92,101 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
...
+
+  constructor(callback: (event: IEvents['onClick']) => void, button?: ActionButton, showPointerFeeback?: boolean, hoverText?: string, interactionDistance?: number) {
     super(callback)
     // This injection is necessary ONLY in events that are ALWAYS turned on and are
     // not assignable to entities. Like events for the UI elements
 
     // TODO(Brian): This will be removed when UI gets back to the entity parenting.
     uuidEventSystem.handlerMap[this.uuid] = this
+
+    if (showPointerFeeback !== undefined) {
```
Comment (pbosio) [inline kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts]: Isn't it better to do `if (showPointerFeeback != null)` in this case? to check for both undefined and null? Question for the Javascript masters on the team like @moliva

Comment (pravusjif) [reply kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts]: I talked with Juanca and Dani and it sounds better to check exclusively for 'false' as the default value is true

**Review by pbosio — APPROVED** [review]

File: kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts [hunk 2]
```diff
@@ -92,52 +92,117 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
...
+  constructor(callback: (event: IEvents['onClick']) => void, options?: any) {
     super(callback)
     // This injection is necessary ONLY in events that are ALWAYS turned on and are
     // not assignable to entities. Like events for the UI elements
 
     // TODO(Brian): This will be removed when UI gets back to the entity parenting.
     uuidEventSystem.handlerMap[this.uuid] = this
+
+    if (options) {
+      this.showFeeback = options.showFeeback === false ? false : true
```
Comment (moliva) [inline kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts]:
```suggestion
      this.showFeeback = !(options.showFeeback === false)
```

File: kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts [hunk 3]
```diff
@@ -92,52 +92,117 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
...
+@Component('engine.onClick', CLASS_ID.UUID_CALLBACK)
+export class OnClick extends OnPointerUUIDEvent<'onClick'> {
   @ObservableComponent.readonly
-  readonly type: string = 'onChange'
-  constructor(callback: (event: IEvents['onChange']) => void) {
+  readonly type: string = 'onClick'
+
+  constructor(callback: (event: IEvents['onClick']) => void)
+  constructor(callback: (event: IEvents['onClick']) => void, options: OnPointerUUIDEventOptions)
+  constructor(callback: (event: IEvents['onClick']) => void, options?: any) {
```
Comment (moliva) [inline kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts:112]: To be consistent with the implementation of `OnPointerDown`.
```suggestion
  constructor(callback: (event: IEvents['onClick']) => void, options?: OnPointerUUIDEventOptions) {
```

File: kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts [hunk 4]
```diff
@@ -92,52 +92,117 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
...
+  constructor(callback: (event: IEvents['pointerDown']) => void, options?: OnPointerUUIDEventOptions) {
     super(callback)
     // This injection is necessary ONLY in events that are ALWAYS turned on and are
     // not assignable to entities. Like events for the UI elements
 
     // TODO(Brian): This will be removed when UI gets back to the entity parenting.
     uuidEventSystem.handlerMap[this.uuid] = this
+
+    if (options) {
+      this.showFeeback = options.showFeeback === false ? false : true
```
Comment (moliva) [inline kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts]: Same as above:
```suggestion
      this.showFeeback = !(options.showFeeback === false)
```

File: kernel/public/test-scenes/135.123.shark-animation/game.ts [hunk 5]
```diff
@@ -1,5 +1,5 @@
-import { Entity, GLTFShape, engine, Vector3, Transform, AnimationState, Animator } from 'decentraland-ecs/src'
-import { OnClick } from 'decentraland-ecs/src/decentraland/UIEvents'
+import { Entity, GLTFShape, engine, Vector3, Transform, AnimationState, Animator, ActionButton } from '../../../packages/decentraland-ecs/src'
```
Comment (moliva) [inline kernel/public/test-scenes/135.123.shark-animation/game.ts]:
```suggestion
import { Entity, GLTFShape, engine, Vector3, Transform, AnimationState, Animator, ActionButton } from 'decentraland-ecs/src'
```

File: kernel/public/test-scenes/135.123.shark-animation/game.ts [hunk 6]
```diff
@@ -1,5 +1,5 @@
...
+import { OnClick } from '../../../packages/decentraland-ecs/src/decentraland/UIEvents'
```
Comment (moliva) [inline kernel/public/test-scenes/135.123.shark-animation/game.ts]:
```suggestion
import { OnClick } from 'decentraland-ecs/src/decentraland/UIEvents'
```

File: kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts [hunk 7]
```diff
@@ -92,52 +92,117 @@ export class OnEnter extends OnUUIDEvent<'onEnter'> {
...
+  constructor(callback: (event: IEvents['pointerUp']) => void, options?: OnPointerUUIDEventOptions) {
     super(callback)
     // This injection is necessary ONLY in events that are ALWAYS turned on and are
     // not assignable to entities. Like events for the UI elements
 
     // TODO(Brian): This will be removed when UI gets back to the entity parenting.
     uuidEventSystem.handlerMap[this.uuid] = this
+
+    if (options) {
+      this.showFeeback = options.showFeeback === false ? false : true
```
Comment (moliva) [inline kernel/packages/decentraland-ecs/src/decentraland/UIEvents.ts]: Same as above:
```suggestion
      this.showFeeback = !(options.showFeeback === false)
```

**Review by moliva — APPROVED** [review]

**Pull request merged**